server/src/modules/clubs/mappers/club-map.ts

```diff
@@ -1,21 +1,78 @@
 import { UniqueEntityID } from '../../../shared/domain/unique-entity-id'
 import { ClubEntity } from '../../../shared/infra/db/entities/club.entity'
-import { Club } from '../domain/entities/club'
-import { ClubDTO } from './club-dto'
+import { EventEntity } from '../../../shared/infra/db/entities/event.entity'
+import { BasicEvent, Club } from '../domain/entities/club'
+
+interface BasicEventDTO {
+  name: string
+  startTime: string
+  endTime: string
+  backgroundImageURL?: string
+  tags: Array<string>
+}
+
+export interface ClubDTO {
+  name: string
+  description: string
+  backgroundImageURL?: string
+  iconURL: string
+  facebookLink?: string
+  twitterLink?: string
+  instagramLink?: string
+  websiteLink?: string
+  tags: Array<string>
+  events: Array<BasicEventDTO>
+}
+
+const basicEventMap = async (events: Array<EventEntity>): Promise<Array<BasicEvent>> => {
+  const eventPromises = events.map(async (event) => {
+    await event.tags.init()
+    return {
+      name: event.name,
+      startTime: event.startTime,
+      endTime: event.endTime,
+      backgroundImageURL: event.backgroundImageURL,
+      tags: event.tags.getItems().map((tag) => tag.name),
+    } as BasicEvent
```

Comment:
Is there any type-safe way to do this?

Reply:
I changed the definition for Basic Event to have the optional fields match that of the Entity definition, and that let me remove the type cast. A TODO we discussed was on having nullable types rather than undefined types, but MikroORM isn't happy about that.

@Property({ nullable: true })
  facebookLink: string

Property 'facebookLink' has no initializer and is not definitely assigned in the constructor.

A cursed way:

@Property({ nullable: true })
  facebookLink!: string | null

The compiler doesn't get mad, but this is pretty confusing.